File: fix_en_structure.js
const fs = require('fs');

// Read the current English translation file
const enPath = './src/locales/en/translation.json';
const enData = JSON.parse(fs.readFileSync(enPath, 'utf8'));

console.log('Current structure - Top level keys:', Object.keys(enData));
console.log('Keys inside landing:', Object.keys(enData.landing || {}));

// The problem: everything is nested inside 'landing'
// We need to extract the correct sections and restructure

// Extract the landing.hero auth-related content to create a proper auth section
const authSection = {
    welcome: "Welcome to PerLeap",
    tagline: "Transform education with AI-powered teaching and personalized learning experiences",
    backToHome: "Back to Home",
    subtitle: "Sign in to your account or create a new one",
    signIn: enData.landing.hero.signIn || "Sign In",
    signUp: enData.landing.hero.signUp || "Sign Up",
    email: enData.landing.hero.email || "Email",
    emailPlaceholder: enData.landing.hero.emailPlaceholder || "you@example.com",
    password: enData.landing.hero.password || "Password",
    signInButton: enData.landing.hero.signInButton || "Sign In",
    signInWithGoogle: enData.landing.hero.signInWithGoogle || "Sign in with Google",
    createAccount: enData.landing.hero.createAccount || "Create Account",
    signUpWithGoogle: enData.landing.hero.signUpWithGoogle || "Sign up with Google",
    orContinueWith: enData.landing.hero.orContinueWith || "Or continue with",
    orSignUpWith: enData.landing.hero.orSignUpWith || "Or sign up with",
    iAmA: enData.landing.hero.iAmA || "I am a...",
    teacher: enData.landing.hero.teacher || "Teacher",
    student: enData.landing.hero.student || "Student",
    selectLanguage: enData.landing.hero.selectLanguage || "Select your preferred language",
    english: enData.landing.hero.english || "English",
    hebrew: enData.landing.hero.hebrew || "עברית (Hebrew)",
    errors: enData.landing.hero.errors || {},
    errors2: enData.landing.hero.errors2 || {},
    success: enData.landing.hero.success || {}
};

// Create proper landing section (only nav and hero with landing-specific content)
const landingSection = {
    nav: enData.landing.nav,
    hero: {
        title1: "Agentic AI for",
        title2: "Education",
        subtitle: "Empower your institution with intelligent agents that automate grading, personalize learning, and transform the educational experience.",
        getStarted: "Get Started",
        learnMore: "Learn More",
        badge: "Introducing Perleap AI 2.0",
        contactUs: "Contact Us"
    },
    mission: "At Perleap, we empower teachers to create intelligent, personalized AI agents, built to reflect their voice, goals, and style in every classroom.",
    features: enData.landing.features || {},
    stats: enData.landing.stats || {},
    cta: enData.landing.cta || {},
    footer: enData.landing.footer || {}
};

// Now extract all other sections that were incorrectly nested in landing
const correctedStructure = {
    landing: landingSection,
    auth: authSection,
    studentDashboard: enData.landing.studentDashboard,
    teacherDashboard: enData.landing.teacherDashboard,
    assignmentChat: enData.landing.assignmentChat,
    calendar: enData.landing.calendar,
    createClassroom: enData.landing.createClassroom,
    editClassroom: enData.landing.editClassroom,
    createAssignment: enData.landing.createAssignment,
    editAssignment: enData.landing.editAssignment,
    classroomDetail: enData.landing.classroomDetail,
    submissionsTab: enData.landing.submissionsTab,
    submissionCard: enData.landing.submissionCard,
    submissionDetail: enData.landing.submissionDetail,
    assignmentDetail: enData.landing.assignmentDetail,
    studentClassroom: enData.landing.studentClassroom,
    settings: enData.landing.settings,
    analytics: enData.landing.analytics,
    wellbeing: enData.landing.wellbeing,
    onboarding: enData.landing.onboarding,
    common: enData.landing.common,
    dimensions: enData.landing.dimensions,
    cra: enData.landing.cra,
    studentOnboarding: enData.landing.studentOnboarding,
    teacherOnboarding: enData.landing.teacherOnboarding,
    studentAnalytics: enData.landing.studentAnalytics,
    classroomAnalytics: enData.landing.classroomAnalytics,
    aboutUs: enData.landing.aboutUs,
    assignments: enData.landing.assignments,
    notifications: enData.landing.notifications,
    assignmentTypes: enData.landing.assignmentTypes
};

console.log('\nNew structure - Top level keys:', Object.keys(correctedStructure));
console.log('Total top-level keys:', Object.keys(correctedStructure).length);

// Write the corrected structure
fs.writeFileSync(enPath, JSON.stringify(correctedStructure, null, 2));
console.log('\n✅ English translation file has been restructured!');

File: fix_en_structure.py
import json

# Read the current English translation file
with open('./src/locales/en/translation.json', 'r', encoding='utf-8') as f:
    en_data = json.load(f)

print('Current structure - Top level keys:', list(en_data.keys()))
print('Keys inside landing:', list(en_data.get('landing', {}).keys())[:10], '...')

# Extract auth-related content from landing.hero to create a proper auth section
hero = en_data['landing'].get('hero', {})
auth_section = {
    "welcome": "Welcome to PerLeap",
    "tagline": "Transform education with AI-powered teaching and personalized learning experiences",
    "backToHome": "Back to Home",
    "subtitle": "Sign in to your account or create a new one",
    "signIn": hero.get('signIn', 'Sign In'),
    "signUp": hero.get('signUp', 'Sign Up'),
    "email": hero.get('email', 'Email'),
    "emailPlaceholder": hero.get('emailPlaceholder', 'you@example.com'),
    "password": hero.get('password', 'Password'),
    "signInButton": hero.get('signInButton', 'Sign In'),
    "signInWithGoogle": hero.get('signInWithGoogle', 'Sign in with Google'),
    "createAccount": hero.get('createAccount', 'Create Account'),
    "signUpWithGoogle": hero.get('signUpWithGoogle', 'Sign up with Google'),
    "orContinueWith": hero.get('orContinueWith', 'Or continue with'),
    "orSignUpWith": hero.get('orSignUpWith', 'Or sign up with'),
    "iAmA": hero.get('iAmA', 'I am a...'),
    "teacher": hero.get('teacher', 'Teacher'),
    "student": hero.get('student', 'Student'),
    "selectLanguage": hero.get('selectLanguage', 'Select your preferred language'),
    "english": hero.get('english', 'English'),
    "hebrew": hero.get('hebrew', 'עברית (Hebrew)'),
    "errors": hero.get('errors', {}),
    "errors2": hero.get('errors2', {}),
    "success": hero.get('success', {})
}

# Create proper landing section (only nav, hero with landing content, mission, features, stats, cta, footer)
landing_section = {
    "nav": en_data['landing'].get('nav', {}),
    "hero": {
        "title1": "Agentic AI for",
        "title2": "Education",
        "subtitle": "Empower your institution with intelligent agents that automate grading, personalize learning, and transform the educational experience.",
        "getStarted": "Get Started",
        "learnMore": "Learn More",
        "badge": "Introducing Perleap AI 2.0",
        "contactUs": "Contact Us"
    },
    "mission": "At Perleap, we empower teachers to create intelligent, personalized AI agents, built to reflect their voice, goals, and style in every classroom.",
    "features": en_data['landing'].get('features', {}),
    "stats": en_data['landing'].get('stats', {}),
    "cta": en_data['landing'].get('cta', {}),
    "footer": en_data['landing'].get('footer', {})
}

# Build the corrected structure with all sections at root level
corrected_structure = {
    "landing": landing_section,
    "auth": auth_section
}

# Extract all other sections from landing and add them as top-level keys
sections_to_extract = [
    'studentDashboard', 'teacherDashboard', 'assignmentChat', 'calendar',
    'createClassroom', 'editClassroom', 'createAssignment', 'editAssignment',
    'classroomDetail', 'submissionsTab', 'submissionCard', 'submissionDetail',
    'assignmentDetail', 'studentClassroom', 'settings', 'analytics', 'wellbeing',
    'onboarding', 'common', 'dimensions', 'cra', 'studentOnboarding',
    'teacherOnboarding', 'studentAnalytics', 'classroomAnalytics', 'aboutUs',
    'assignments', 'notifications', 'assignmentTypes'
]

for section in sections_to_extract:
    if section in en_data['landing']:
        corrected_structure[section] = en_data['landing'][section]

print('\nNew structure - Top level keys:', list(corrected_structure.keys()))
print('Total top-level keys:', len(corrected_structure.keys()))

# Write the corrected structure
with open('./src/locales/en/translation.json', 'w', encoding='utf-8') as f:
    json.dump(corrected_structure, f, ensure_ascii=False, indent=2)

print('\n✅ English translation file has been restructured!')
